Modules/Loadable/Segmentations/qSlicerSegmentationsModuleWidget.cxx

```diff
@@ -950,12 +954,42 @@ bool qSlicerSegmentationsModuleWidget::exportFromCurrentSegmentation()
   // Do the export
   if (labelmapNode)
     {
+
+    vtkMRMLVolumeNode* referenceVolumeNode = vtkMRMLVolumeNode::SafeDownCast(d->MRMLNodeComboBox_ExportLabelmapReferenceVolume->currentNode());
+    if (referenceVolumeNode)
+      {
+      vtkNew<vtkStringArray> segmentIDsArray;
+      segmentIDsArray->SetNumberOfValues(segmentIDs.size());
+      for (int i = 0; i < segmentIDs.size(); ++i)
+        {
+        segmentIDsArray->SetValue(i, segmentIDs[i]);
+        }
+
+      if (vtkSlicerSegmentationsModuleLogic::IsEffectiveExentOutsideReferenceVolume(
+        referenceVolumeNode, d->SegmentationNode, segmentIDsArray))
+        {
+        ctkMessageBox* exportWarningMesssgeBox = new ctkMessageBox(this);
+        exportWarningMesssgeBox->setAttribute(Qt::WA_DeleteOnClose);
+        exportWarningMesssgeBox->setWindowTitle("Export may erase data");
+        exportWarningMesssgeBox->addButton(QMessageBox::StandardButton::Ok);
+        exportWarningMesssgeBox->addButton(QMessageBox::StandardButton::Cancel);
+        exportWarningMesssgeBox->setDontShowAgainVisible(true);
+        exportWarningMesssgeBox->setIcon(QMessageBox::Warning);
+        exportWarningMesssgeBox->setDontShowAgainSettingsKey("Segmentations/AlwaysCropDuringSegmentationNodeExport");
+        exportWarningMesssgeBox->setText("Reference volume does not cover the whole segmentation. "
```

Comment:
I'm not sure if this message is clear enough for everyone. What do you think about something like this?
"The current segmentation does not completely fit into the new geometry. Do you want to crop the segmentation?"

Reply:
Sure that message sounds good to me.